### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

**Review by ashtul — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
```
Comment (chayim) [inline .github/workflows/codeql-analysis.yml:19]: Recommend you add versioned branches        - '[0-9]+.[0-9]+'

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
```
Comment (chayim) [inline .github/workflows/codeql-analysis.yml:16]: Recommend you add versioned branches        - '[0-9]+.[0-9]+'

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  pull_request:
```
Comment (chayim) [inline .github/workflows/codeql-analysis.yml:17]: We can remove the pull request action check if we maintain an on-push to versioned branches

**Review by chayim — CHANGES_REQUESTED** [review]
See comments re versioned branches.

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (gkorland) [reply .github/workflows/codeql-analysis.yml:17]: with pull request you can check the code before you merge it

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (gkorland) [reply .github/workflows/codeql-analysis.yml:16]: I tried it, but it seems like they don't know how to parse it...

**Review by chayim — APPROVED** [review]

**Pull request merged**